### PR TITLE
Add test for apiserver with kubeconfig

### PR DESCRIPTION
**Pull request by kd7lxl**
This adds a helm test pod that connects to apiserver using the generated kubeconfig and runs a sample command we always expect to succeed, `kubectl get ns`. This serves as both an integration test and as an example for konk users. It is exercised by Github Actions CI to prove konk works.

### CODE REVIEW

**Review by abalaven — APPROVED** [review]

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -14,4 +14,13 @@ jobs:
...
         run: |
-          make helm-lint HELM="docker run --rm -i -v $PWD:/apps infoblox/helm:3.2.4-5b243a2"
+          mkdir -p ~/.kube
+          make helm-lint
+      - uses: helm/kind-action@v1.0.0
+      - name: Test
+        working-directory: ${{ github.workspace }}/src/github.com/${{ github.repository }}
+        run: |
+          kubectl cluster-info
+          make deploy-konk
```
Comment (drewwells) [inline .github/workflows/pr.yaml:24]: Why isn't this using --wait?

File: charts/konk/templates/tests/test-connection.yaml [hunk 2]
```diff
@@ -8,8 +8,27 @@ metadata:
...
-      image: busybox
-      command: ['wget']
-      args: ['{{ include "konk.fullname" . }}:{{ .Values.service.port }}']
+  - name: kind
+    image: "{{ .Values.kind.image.repository }}:{{ .Values.kind.image.tag | default .Chart.AppVersion }}"
+    imagePullPolicy: {{ .Values.kind.image.pullPolicy }}
+    command:
+      - kubectl
+    args:
+      - --server=https://{{ include "konk.fullname" . }}:{{ .Values.service.port }}
```
Comment (drewwells) [inline charts/konk/templates/tests/test-connection.yaml:17]: Todo default kubeconfig to fqdn service name

File: charts/konk/templates/deployment.yaml [hunk 3]
```diff
@@ -37,19 +38,37 @@ spec:
...
+              path: /healthz
+              port: https
+              scheme: HTTPS
+          {{- if .Values.apiserver.startupProbe }}
+          startupProbe:
+            httpGet:
+              path: /healthz
+              port: https
+              scheme: HTTPS
+          {{- end }}
```
Comment (drewwells) [inline charts/konk/templates/deployment.yaml:63]: Neat!

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (kd7lxl) [reply .github/workflows/pr.yaml:24]: The static init pod never becomes ready, so `--wait` blocks forever. We can add `--wait` once this issue is resolved.

**Review by drewwells — APPROVED** [review]

File: charts/konk/templates/tests/test-connection.yaml (hunk 2 above)
Comment (kd7lxl) [reply charts/konk/templates/tests/test-connection.yaml:17]: Completed in https://github.com/infobloxopen/konk/pull/16

**Pull request merged**